utils/src/const_value.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    fmt::{Debug, Display},
+    hash::{Hash, Hasher},
+    ops::Deref,
+};
+
+/// A wrapper for a type that is not supposed to be modified, no matter one
+/// This particularly solves the problem of not being able to mark member variables as const in Rust
+pub struct ConstValue<T> {
+    value: T,
+}
+
+impl<T> ConstValue<T> {
+    pub fn new(value: T) -> Self {
+        Self { value }
+    }
+}
```

Comment:
Should there also be a method to take the inner value out of the `ConstValue` wrapper?

Reply:
Done.